packages/hardhat/contracts/Future.sol

```diff
@@ -50,23 +44,24 @@ contract Future is IUniswapV3SwapCallback {
         }
     }
 
-    function long(uint quantity, uint price) external returns (uint amountReceived) {
+    function long(int quantity, uint price) external override returns (int amountReceived, int amountPaid) {
```

Comment:
why did you change from uint to int? can this number be negative? should we check these parameters?

Reply:
I think it was to avoid some casting, I have a full review of the types and params in my todo list